flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -816,7 +816,20 @@ class OmpAttributeVisitor : DirectiveAttributeVisitor<llvm::omp::Directive> {
   const parser::OmpClause *GetAssociatedClause() { return associatedClause; }
 
 private:
+  std::int64_t SetAssociatedMaxClause(llvm::SmallVector<std::int64_t> &,
+      llvm::SmallVector<const parser::OmpClause *> &);
+  std::int64_t GetAssociatedLoopLevelFromLoopConstruct(
+      const parser::OpenMPLoopConstruct &);
   std::int64_t GetAssociatedLoopLevelFromClauses(const parser::OmpClauseList &);
+  void CollectAssociatedLoopLevelsFromLoopConstruct(
+      const parser::OpenMPLoopConstruct &, llvm::SmallVector<std::int64_t> &,
+      llvm::SmallVector<const parser::OmpClause *> &);
+  void CollectAssociatedLoopLevelsFromInnerLoopContruct(
+      const parser::OpenMPLoopConstruct &, llvm::SmallVector<std::int64_t> &,
+      llvm::SmallVector<const parser::OmpClause *> &);
+  void CollectAssociatedLoopLevelsFromClauses(const parser::OmpClauseList &,
+      llvm::SmallVector<std::int64_t> &,
+      llvm::SmallVector<const parser::OmpClause *> &);
```

Comment:
If these do not require `OmpAttributeVisitor`, consider making them standalone static functions.

Reply:
We're using some members in the class currently. I think I tried making them static functions initially, but it didn't work out very well.